sympy/matrices/expressions/matexpr.py

```diff
@@ -457,20 +457,13 @@ class Identity(MatrixExpr):
 
     is_Identity = True
 
-    def __new__(cls, n):
-        return super(Identity, cls).__new__(cls, sympify(n))
-
-    @property
-    def rows(self):
-        return self.args[0]
-
-    @property
-    def cols(self):
-        return self.args[0]
+    def __new__(cls, n, m = None):
+        if m is None: m = n
+        return super(Identity, cls).__new__(cls, sympify(n), simplify(m))
```

Comment:
Why are you adding another argument? (And if it is necessary -- and I"m not convinced it is -- it should be `sympify` not `simplify`.)

Reply:
Identity matrix should be square, so you shouldn't be adding another argument, right?

Reply:
right, thank's :)

I swapped something with function eye e.g. in matlab which allowed eye(n,m)